File: Sdl.Tridion.Context/BrowserClaims.cs
using System.Collections.Generic;
using Tridion.ContentDelivery.AmbientData;

namespace Sdl.Tridion.Context
{
    public class BrowserClaims
    {

        public BrowserClaims()
        {
        }

        public bool CookieSupport
        {
            get
            {
                // BUG: Returns false on desktop FF, IE, Safari
                return AmbientDataContext.CurrentClaimStore.Get<bool>(ClaimUris.UriCookieSupport);
            }
        }

        public string CssVersion { get { return AmbientDataContext.CurrentClaimStore.Get<string>(ClaimUris.UriCssVersion); } }

        public int DisplayColorDepth { get { return AmbientDataContext.CurrentClaimStore.Get<int>(ClaimUris.UriDisplayColorDepth); } }

        public int DisplayHeigth { get { return AmbientDataContext.CurrentClaimStore.Get<int>(ClaimUris.UriBrowserDisplayHeight); } }

        public int DisplayWidth { get { return AmbientDataContext.CurrentClaimStore.Get<int>(ClaimUris.UriBrowserDisplayWidth); } }

        public List<string> ImageFormatSupport
        {
            get
            {
                return AmbientDataContext.CurrentClaimStore.Get<List<string>>(ClaimUris.UriImageFormatSupport);
            }
        }

        public List<string> InputDevices
        {
            get
            {
                return AmbientDataContext.CurrentClaimStore.Get<List<string>>(ClaimUris.UriInputDevices);
            }
        }

        public List<string> InputModeSupport
        {
            get { return AmbientDataContext.CurrentClaimStore.Get<List<string>>(ClaimUris.UriInputModeSupport); }
        }

        public string JsVersion { get { return AmbientDataContext.CurrentClaimStore.Get<string>(ClaimUris.UriJsVersion); } }

        public List<string> MarkupSupport
        {
            get { return AmbientDataContext.CurrentClaimStore.Get<List<string>>(ClaimUris.UriMarkupSupport); }
        }
        public string Model
        {
            get { return AmbientDataContext.CurrentClaimStore.Get<string>(ClaimUris.UriBrowserModel); }
        }

        public string PreferredHtmlContentType { get { return AmbientDataContext.CurrentClaimStore.Get<string>(ClaimUris.UriPreferredHtmlContentType); } }

        public List<string> ScriptSupport { get { return AmbientDataContext.CurrentClaimStore.Get<List<string>>(ClaimUris.UriScriptSupport); } }

        public List<string> StylesheetSupport
        {
            get { return AmbientDataContext.CurrentClaimStore.Get<List<string>>(ClaimUris.UriStylesheetSupport); }
        }

        public string Variant { get { return AmbientDataContext.CurrentClaimStore.Get<string>(ClaimUris.UriBrowserVariant); } }

        public string Vendor { get { return AmbientDataContext.CurrentClaimStore.Get<string>(ClaimUris.UriBrowserVendor); } }

        public string Version { get { return AmbientDataContext.CurrentClaimStore.Get<string>(ClaimUris.UriBrowserVersion); } }


    }
}

File: com/sdl/tridion/context/BrowserClaims.java
package com.sdl.tridion.context;

import java.net.URI;
import java.util.Map;

public class BrowserClaims extends ContextClaims {

	public BrowserClaims(Map<URI, Object> claims) {
		super(claims);
		tafContext = "taf:claim:context:browser:";
	}

	public BrowserClaims() {
		super();
		tafContext = "taf:claim:context:browser:";
	}

	public boolean hasCookieSupport() {
		return getBooleanValue(ClaimUris.UriCookieSupport);
	}
	public String getCssVersion() {
		return getStringValue(ClaimUris.UriCssVersion);
	}

	public int getDisplayColorDepth() {
		return getIntValue(ClaimUris.UriDisplayColorDepth);
	}

	public int getDisplayHeigth() {
		return getIntValue(ClaimUris.UriBrowserDisplayHeight);
	}

	public int getDisplayWidth() {
		return getIntValue(ClaimUris.UriBrowserDisplayWidth);
	}

	public String getImageFormatSupport() {
		return getStringValue(ClaimUris.UriImageFormatSupport);
	}

	public String getInputDevices() {
		return getStringValue(ClaimUris.UriInputDevices);
	}

	public String getInputModeSupport() {
		return getStringValue(ClaimUris.UriInputModeSupport);
	}

	public String getJsVersion() {
		return getStringValue(ClaimUris.UriJsVersion);
	}

	public String getMarkupSupport() {
		return getStringValue(ClaimUris.UriMarkupSupport);
	}

	public String getModel() {
		return getStringValue(ClaimUris.UriBrowserModel);
	}

	public String getPreferredHtmlContentType() {
		return getStringValue(ClaimUris.UriPreferredHtmlContentType);
	}

	public String getScriptSupport() {
		return getStringValue(ClaimUris.UriScriptSupport);
	}

	public String getStylesheetSupport() {
		return getStringValue(ClaimUris.UriStylesheetSupport);
	}

	public String getVariant() {
		return getStringValue(ClaimUris.UriBrowserVariant);
	}

	public String getVendor() {
		return getStringValue(ClaimUris.UriBrowserVendor);
	}

	public String getVersion() {
		return getStringValue(ClaimUris.UriBrowserVersion);
	}

}
